core/server/master/src/main/java/alluxio/master/meta/MetaMasterSync.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.heartbeat.HeartbeatExecutor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Task that carries out the necessary standby master to leader master communications, including
+ * register and heartbeat. This class manages its own {@link MetaMasterMasterClient}.
+ *
+ * When running, this task first requests master configuration from the standby master,
+ * then sends it to the leader master. After which, the task will wait for the elapsed time
+ * since its last heartbeat has reached the heartbeat interval. Then the cycle will continue.
+ *
+ * If the task fails to heartbeat to the leader master, it will destroy its old master client
```

Comment:
is this necessary? I think the `MetaMasterClient` should be able to handle the retries.

Reply:
I think this line can be deleted, but it is the executor service that handle the retries.

```
  private final ExecutorService mExecutorService;
  protected void startMasters(boolean isLeader) {
    try {
      mWorkerConfigChecker.reset();
      if (isLeader) {
        mSafeModeManager.notifyPrimaryMasterStarted();

        //  The service that detects lost standby master nodes
        mExecutorService.submit(new HeartbeatThread(
            HeartbeatContext.MASTER_LOST_MASTER_DETECTION,
            new LostMasterDetectionHeartbeatExecutor(),
            (int) Configuration.getMs(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
      } else {
        // Standby master should setup MetaMasterSync to communicate with the leader master
        mMetaMasterSync = new MetaMasterSync(mMasterId, mMasterHostname, mMetaMasterClient);
        mExecutorService.submit(new HeartbeatThread(HeartbeatContext.MASTER_SYNC, mMetaMasterSync,
                (int) Configuration.getMs(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
      }
```

Reply:
The heartbeat will eventually provide retries since it keeps running periodically, but I think the client itself also provides retries: https://github.com/Alluxio/alluxio/pull/7233/files#diff-8495e238f9f1a6a00192d61f41daddd9R89

Reply:
I see. Both the executor service and the client itself provide retries. I will delete this line.